polynote-kernel/src/main/scala/polynote/kernel/lang/python/PythonInterpreter.scala

```diff
@@ -32,60 +29,31 @@ class PythonInterpreter(val symbolTable: RuntimeSymbolTable) extends LanguageInt
 
   override def shutdown(): IO[Unit] = shutdownSignal.complete.flatMap(_ => withJep(jep.close()))
 
-  override def init(): IO[Unit] = {
-    implicit val s: ContextShift[IO] = listenerShift
-
-      // note: leaving old TODOs in case they might still be useful?
-      // TODO: The downside of doing this eagerly is that Python code won't forbid using symbols defined in later cells.
-      //       Should that be fixed? Maybe we could build a new locals dict during runCode?
-      // TODO: Can there be some special treatment for function values to make them Callable in python?
-      // TODO: Instead of doing this, can we somehow just make the symbol table a Python scope?
-
-    // watch for new symbols and push 'em to Jep
-    symbolTable.subscribe(Option(this)) {
-      value => withJep {
-        value.value match {
-          case polynote.runtime.Runtime => // pass (already handled above)
-          case _ =>
-            jep.set(value.name, value.value)
-        }
-      }
-    }.interruptWhen(shutdownSignal()).compile.drain.unsafeRunAsyncAndForget()
-
-    // make sure to grab any symbols that have already been created
-    withJep {
-      val terms = symbolTable.currentTerms
-      terms.foreach {
-        value =>
-          value.value match {
-            case polynote.runtime.Runtime =>
-              // hijack the kernel and wrap it in a pyobject so we can set the display
-              // we need to do this because it looks like jep doesn't handle the Runtime.display object properly
-              // (maybe because it doesn't fully support Scala).
-              // we need to create a new Python class that proxies the runtime object so that we can use
-              // grab it as a PyObject from jep. We need a PyObject so we can use setAttr.
-              // Then we can set the `display` attribute to the display object ourselves.
```

Comment:
Hmm, this comment is a bit overly verbose (wonder who wrote it... 🤔) but I think something explaining a bit about what we're doing here would still be useful.

Reply:
Added back some explanation about it in #191